docs/services/invocation.md

```diff
@@ -141,38 +141,60 @@ The Invocation identifier is opaque and its current format should not be relied
 
 ## Cancel an invocation
 
-:::caution
+If an invocation takes too long to complete and, therefore, is no longer of interest, you can cancel it.
+Canceling an invocation allows it to free any resources it is holding and roll back any changes it has made so far.
+In order to roll back changes correctly, the service handlers need to contain the necessary compensation logic.
+If the required compensation logic is implemented, then the service state stays consistent even in the presence of cancellations.
 
-At the moment, gracefully cancelling an invocation is not supported. It will be supported in future Restate releases.
+The cancellation process works recursively in the following way:
+First, Restate tries to cancel the leaves of the current invocation, i.e. complete pending sleeps and awakeables or try to cancel calls to other services.
+Once the leaves are canceled, the service handler will be notified about the cancellation via a terminal error being thrown at the call site.
+This allows the handler to run its specific compensation logic.
+The response of the handler will then be propagated back to its caller where it will continue with the cancellation process.
 
+:::note
+Canceling an invocation is a non-blocking operation which in some rare cases needs to be retried by the user.
```

Comment:
I don't really understand what is meant here.

Reply:
Cancellation is a non-blocking operation which means that it returns without the operation being complete. Sometimes cancellations have no effect which means that users have to trigger the cancellation again. If this is more helpful, then I'll update the part.